### PR TITLE
Java 21 Migration

### LINKED ISSUE #575: Java 21 test fails
The current codebase compiles on the current LTS release Java 21 but there are 791 failing test cases
1. Gradle 8.4 is incompatible with Java 21.
2. JPF Core relies on `jdk.internal.*` and `SharedSecrets` accessors that were removed or refactored in Java 21.
3. Introduction of virtual threads and Thread Container

Image

I am testing on the Java-17 branch as following: https://gist.github.com/Sameersawant77/5649e23e4d6ddc599c64c512af63f5d2

### PR DESCRIPTION
**Pull request by Sameersawant77**
Fixes #575 

Refactoring and Migrating the codebase to support Java 21
I have reduced test failures from 791 down to 9. I am currently cleaning up the remaining edge cases and will open a PR shortly.

Changes:
Gradle Upgrade
Native Peers
Internal Accessors
Virtual Threads and Threads Container Handling

image

### CODE REVIEW

File: src/classes/modules/java.base/jdk/internal/access/SharedSecrets.java [hunk 1]
```diff
@@ -17,10 +17,9 @@
  */
 package jdk.internal.access;
 import jdk.internal.misc.Unsafe;
-import java.io.File;
-import java.io.FileDescriptor;
-import java.io.ObjectInputFilter;
-import java.io.ObjectInputStream;
+
+import java.io.*;
```
Comment (cyrille-artho) [inline src/classes/modules/java.base/jdk/internal/access/SharedSecrets.java]: Using wildcards is convenient but obscures which classes we actually import; please keep the explicit list of classes as before this change.

**Review by cyrille-artho — CHANGES_REQUESTED** [review]
Great work!
Looks good overall; I have only a few minor comments.
However, please make your PR against branch `java-21`. I have created it remotely now.
We are still using `java-17` to support Java 17, and we have all tests passing there now.

**Comment by cyrille-artho** [comment]
Looks good; please make a new PR with the minor changes I've requested to branch `java-21`. This way, we can keep support for Java 17 (hopefully soon with extensions, too).
